test/powershell/Language/Parser/Parser.Tests.ps1

```diff
@@ -915,4 +915,9 @@ foo``u{2195}abc
         # Issue #2780
         { ExecuteCommand "`$herestr=@`"`n'`"'`n`"@" } | Should Not Throw
     }
+
+    It "Throw better error message when statement should be put in named blocks" {
+        $err = { ExecuteCommand "Function foo { [CmdletBinding()] param() DynamicParam {} Hi" } | Should -Throw -PassThru
```

Comment:
Can you also check the parent errorid: Should -Throw -ErrorId "errorid expected" -PassThru

Reply:
Sure thing. Fixed.